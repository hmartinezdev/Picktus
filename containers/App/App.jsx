import React, { Component } from "react";
import Header from "@components/Header";
import Logo from "@components/Logo";
import colors from "../../constants/colors";

class App extends Component {
  render() {
    return (
      <div className="page">
        <Header />
        <Logo />
        <style jsx>{`
          .page {
            min-height: 100vh;
            background-color: ${colors.main};
          }
          :global(*) {
            padding: 0;
            margin: 0;
          }

          :global(html) {
            font-size: 100%;
          }
        `}</style>
      </div>
    );
  }
}

export default App;
