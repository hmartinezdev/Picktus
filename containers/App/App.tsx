import React, { Component } from 'react';
import Header from '@components/Header';
import colors from '../../constants/colors';

interface PropsType {
  children: JSX.Element[] | JSX.Element;
}
class App extends Component<PropsType, {}> {
  render() {
    const { children } = this.props;

    return (
      <div className="page">
        <Header />
        {children}
        <style jsx>{`
          .page {
            min-height: 100vh;
            background-color: ${colors.primary};
          }
          :global(*) {
            padding: 0;
            margin: 0;
          }

          :global(html) {
            font-size: 100%;
          }
        `}</style>
      </div>
    );
  }
}

export default App;
