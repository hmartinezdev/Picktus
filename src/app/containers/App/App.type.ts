export interface IReduxStateProps {
  showLoader: boolean;
}
