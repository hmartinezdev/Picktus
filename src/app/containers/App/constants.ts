export const config = {
  apiKey: 'AIzaSyAVQ-Abn_Sf0z18jPeW_mKhLOwwGfWn_40',
  authDomain: 'picktus-76101.firebaseapp.com',
  databaseURL: 'https://picktus-76101.firebaseio.com',
  projectId: 'picktus-76101',
  storageBucket: 'picktus-76101.appspot.com',
  messagingSenderId: '1077268436650',
};
