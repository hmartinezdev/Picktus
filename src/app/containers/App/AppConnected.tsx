import { IReduxState } from '@store/reducers';
import { userLoginSuccess } from '@store/reducers/user/user-actions';
import { connect } from 'react-redux';
import App from './App';
import { IReduxStateProps } from './App.type';

const mapStateToProps = (state: IReduxState): IReduxStateProps => {
  return {
    showLoader: state.display.loader,
  };
};

export default connect(
  mapStateToProps,
  { userLoginSuccess }
)(App);
