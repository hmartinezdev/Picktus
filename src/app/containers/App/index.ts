import { IReduxState } from '@store/reducers';
import { connect } from 'react-redux';
import App from './App';
import { IReduxStateProps } from './App.type';

const mapStateToProps = (state: IReduxState): IReduxStateProps => {
  return {
    showLoader: state.display.loader,
  };
};

export default connect(mapStateToProps)(App);
