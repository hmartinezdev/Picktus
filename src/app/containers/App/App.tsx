import Background from '@components/Background';
import Header from '@components/Header';
import Loader from '@components/Loader';
import Messages from '@components/Messages';
import firebase from 'firebase/app';
import { withRouter } from 'next/router';
import React, { Component } from 'react';
import { Transition, TransitionGroup } from 'react-transition-group';
import { IAppPropsType } from './App.type';
import { config } from './constants';

class App extends Component<IAppPropsType> {
  constructor(props: IAppPropsType) {
    super(props);

    // If the firebase setup has not been done yet
    // we do it here
    if (!firebase.apps.length) {
      firebase.initializeApp(config);
    }
  }

  public render(): React.ReactElement<App> {
    const { children, showLoader, router } = this.props;

    return (
      <Background>
        <div className="page">
          <Header />
          <Messages />
          {showLoader && <Loader />}
          <TransitionGroup className="app_transitiongroup" appear={false}>
            {React.Children.map(children, (child) => (
              <Transition key={router.route} timeout={600} mountOnEnter={true} unmountOnExit={true} appear={false}>
                {(status) => React.cloneElement(child as React.ReactElement<any>, { status })}
              </Transition>
            ))}
          </TransitionGroup>
          <style jsx>{`
            .page {
              overflow: hidden;
              min-height: 100vh;
              display: flex;
              align-self: stretch;
              overflow: hidden;
              width: 100%;
            }

            :global(.app_transitiongroup) {
              overflow: hidden;
              min-height: 100vh;
              display: flex;
              align-self: stretch;
              overflow: hidden;
              width: 100%;
            }

            :global(*) {
              padding: 0;
              margin: 0;
            }

            :global(html) {
              font-size: 100%;
            }
          `}</style>
        </div>
      </Background>
    );
  }
}

export default withRouter(App);
