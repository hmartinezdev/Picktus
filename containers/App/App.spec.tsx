import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

let wrapper;

const setup = () => shallow(<App />);

describe('<App />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
