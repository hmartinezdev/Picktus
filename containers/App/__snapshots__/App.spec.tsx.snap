// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render properly 1`] = `
<div
  className="page"
>
  <Header />
  <Logo />
  <style
    jsx={true}
  >
    
          .page {
            min-height: 100vh;
            background-color: #32302e;
          }
          :global(*) {
            padding: 0;
            margin: 0;
          }

          :global(html) {
            font-size: 100%;
          }
        
  </style>
</div>
`;
